examples/example_ppdWrite_segmentFile_test.go

```diff
@@ -82,3 +84,13 @@ func Example_ppdWriteSegmentFile() {
 	// credit 1: 622231380104837098765        0100000000               Credit Account 2        0121042880000003
 	// debit 0: 627231380104123456789        0200000000               Debit Account           0121042880000001
 }
+
+func Test_ppdMixedDebitCreditImmediateOrigin(t *testing.T) {
+	f, err := os.Open(filepath.Join("testdata", "ppd-mixedDebitCredit.ach"))
+	require.NoError(t, err)
+	r := ach.NewReader(f)
+	achFile, err := r.Read()
+	require.NoError(t, err)
+
+	require.Equal(t, achFile.Header.ImmediateOrigin, "0121042882")
```

Comment:
You probably want to use the Field() accessors. 

```suggestion
	require.Equal(t, achFile.Header.ImmediateOriginField(), "0121042882")
```